### PR TITLE
Create GrapesPinTextField composable

### PR DESCRIPTION
**Pull request by kelianClerc**
https://user-images.githubusercontent.com/26220096/234535359-ec4f0668-0aa2-4817-8147-927e78c75ce7.mov

### CODE REVIEW

File: library-compose/src/main/java/com/spendesk/grapes/compose/textfield/GrapesPinTextFieldDecorationBox.kt [hunk 1]
```diff
@@ -0,0 +1,92 @@
...
+            .border(
+                width = GrapesPintTextFieldDefaults.PinCharBorderWidth,
+                color = GrapesPintTextFieldDefaults.computePinCharBorderColor(isEnabled = isEnabled, isError = isError, isFocused = isFocused),
+                shape = GrapesTheme.shapes.small
+            )
+            .pointerInput(Unit) {
+                detectTapGestures(
+                    onPress = { onClick() }
+                )
+            }
```
Comment (jpbarbaud) [inline library-compose/src/main/java/com/spendesk/grapes/compose/textfield/GrapesPinTextFieldDecorationBox.kt:81]: Why not 
```suggestion
.clickable(
    onClick = onClick
)
```

File: library-compose/src/main/java/com/spendesk/grapes/compose/textfield/GrapesPinTextFieldDecorationBox.kt [hunk 2]
```diff
@@ -0,0 +1,92 @@
...
+            .padding(GrapesPintTextFieldDefaults.PinCharPadding),
+        contentAlignment = Alignment.Center
+    ) {
+        Text(
+            text = char,
+            style = GrapesTheme.typography.bodyM,
+            color = GrapesPintTextFieldDefaults.computePinCharTextColor(isEnabled = isEnabled, isError = isError, isFocused = isFocused),
```
Comment (jpbarbaud) [inline library-compose/src/main/java/com/spendesk/grapes/compose/textfield/GrapesPinTextFieldDecorationBox.kt:88]: Could be interesting to manage color like compose `TextFieldColors`.

```kotlin
interface GrapesPinTextColors {
    textColor(enabled: Boolean, error:Boolean, interactionSource: InteractionSource)
    
    backgroundColor(enabled: Boolean, error:Boolean, interactionSource: InteractionSource)
    
    borderColor(enabled: Boolean, error:Boolean, interactionSource: InteractionSource)
}

DefaultsPinTextColor(
   private val textColor: Color,
   private val disabledTextColor: Color,
   private val focusTextColor: Color,
   private val errorTextColor: Color,
   
   etc.
) :  GrapesPinTextColors {


```
textColor(enabled: Boolean, error:Boolean, interactionSource: InteractionSource): State<Color> {
return rememberUpdatedState(
        val focused by interactionSource.collectIsFocusedAsState()

        val targetValue = when {
            !enabled -> disabledTextColor
            error -> errorTextColor
            focused -> focusTextColor
            else -> textColor
        }
        return rememberUpdatedState(targetValue)
}
    
    backgroundColor(enabled: Boolean, error:Boolean, interactionSource: InteractionSource) { … }
    
    borderColor(enabled: Boolean, error:Boolean, interactionSource: InteractionSource) { … }
```

File: library-compose/src/main/java/com/spendesk/grapes/compose/textfield/GrapesPinTextFieldDefaults.kt [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+        isEnabled: Boolean,
+        isError: Boolean,
+        isFocused: Boolean
+    ): Color =
+        when {
+            isEnabled.not() -> GrapesTheme.colors.mainPrimaryLighter
+            isError -> GrapesTheme.colors.mainAlertNormal
+            isFocused && isEnabled -> GrapesTheme.colors.mainPrimaryDark
+            else -> GrapesTheme.colors.mainPrimaryNormal
+        }
```
Comment (jpbarbaud) [inline library-compose/src/main/java/com/spendesk/grapes/compose/textfield/GrapesPinTextFieldDefaults.kt:48]: See my comment about color above

**Review by jpbarbaud — CHANGES_REQUESTED** [review]
1 suggestion about how we handle colors

File: library-compose/src/main/java/com/spendesk/grapes/compose/textfield/GrapesPinTextFieldDecorationBox.kt (hunk 1 above)
Comment (kelianClerc) [reply library-compose/src/main/java/com/spendesk/grapes/compose/textfield/GrapesPinTextFieldDecorationBox.kt:81]: clickable adds a ripple that is ugly

File: library-compose/src/main/java/com/spendesk/grapes/compose/textfield/GrapesPinTextFieldDecorationBox.kt (hunk 2 above)
Comment (kelianClerc) [reply library-compose/src/main/java/com/spendesk/grapes/compose/textfield/GrapesPinTextFieldDecorationBox.kt:88]: isFocused is different from real focus handling, it's rather, doesThePinCorrespondToCursorIndex.
But for the rest ok I'll make the change

**Pull request merged**